src/bokeh/models/annotations/legends.py

```diff
@@ -531,6 +538,28 @@ def js_on_click(self, handler: JsEventCallback) -> None:
         """ Set up a JavaScript handler for legend item clicks. """
         self.js_on_event(LegendItemClick, handler)
 
+class Val(HasProps):
```

Comment:
Is there a way to infer `Val` based on the string only? This adds overhead when creating a `Scalebar` for a subcoordinate plot like the one you have in your example.

Reply:
I think we should try to infer, as we often do. We do not change enum values very often in my recollection and even then a user would have to be using specifically whatever the update enum is... I think it is sufficient to provide something like `Val` that can be used optionally to completely disambiguate if desired, and be clear in the documentation about what enum might conflict.